index.js

```diff
@@ -107,53 +107,61 @@ class SimpleKeyring extends EventEmitter {
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData_v1 (withAccount, typedData, opts = {}) {
-    const privKey = this.getPrivateKeyFor(withAccount, opts);
+  signTypedData_v1 (withAccount, typedData) {
+    const privKey = this.getPrivateKeyFor(withAccount);
     const sig = sigUtil.signTypedDataLegacy(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData_v3 (withAccount, typedData, opts = {}) {
-    const privKey = this.getPrivateKeyFor(withAccount, opts);
+  signTypedData_v3 (withAccount, typedData) {
+    const privKey = this.getPrivateKeyFor(withAccount);
     const sig = sigUtil.signTypedData(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
   // personal_signTypedData, signs data along with the schema
-  signTypedData_v4 (withAccount, typedData, opts = {}) {
-    const privKey = this.getPrivateKeyFor(withAccount, opts);
+  signTypedData_v4 (withAccount, typedData) {
+    const privKey = this.getPrivateKeyFor(withAccount);
     const sig = sigUtil.signTypedData_v4(privKey, { data: typedData })
     return Promise.resolve(sig)
   }
 
-  getPrivateKeyFor (address, opts = {}) {
+  getPrivateKeyFor (address) {
     if (!address) {
       throw new Error('Must specify address.');
     }
-    const wallet = this._getWalletForAccount(address, opts)
+    const wallet = this._getWalletForAccount(address)
     const privKey = ethUtil.toBuffer(wallet.getPrivateKey())
     return privKey;
   }
 
-  // returns an address specific to an app
-  getAppKeyAddress (address, origin) {
-    return new Promise((resolve, reject) => {
-      try {
-        const wallet = this._getWalletForAccount(address, {
-          withAppKeyOrigin: origin,
-        })
-        const appKeyAddress = sigUtil.normalize(wallet.getAddress().toString('hex'))
-        return resolve(appKeyAddress)
-      } catch (e) {
-        return reject(e)
-      }
-    })
+  async getAppKey (address, origin) {
```

Comment:
Thoughts on this method name? Would `getPrivateAppKey` or `exportAppkey` be better?

I was hoping that the "Key" part would suffice for communicating that this is a _private key_, but maybe not.

Reply:
Personally i prefer to see `secret` or `private` (indicating "DANGER"), thus I'd go with `getAppPrivateKey` (`PrivateKey` should be kept whole, thus NOT getPrivateAppKey).